packages/core/src/data-import/providers/importer/importer.ts

```diff
@@ -158,15 +159,29 @@ export class Importer {
                 featuredAssetId: productAssets.length ? productAssets[0].id : undefined,
                 assetIds: productAssets.map(a => a.id),
                 facetValueIds: await this.getFacetValueIds(product.facets, languageCode),
-                translations: [
-                    {
-                        languageCode,
-                        name: product.name,
-                        description: product.description,
-                        slug: product.slug,
-                        customFields,
-                    },
-                ],
+                translations:
+                    !product.translations || product.translations.length === 0
```

Comment:
I commented on the interface about the need for the top-level translatable fields (name, description, slug). If they are indeed not essential, then this part could be simplified to just the `map` operation part. Same would go for the other places using the same pattern below.

Reply:
Yes, getting rid of the whole `mainLanguage` part would simplify things quite a bit. The only worry is backwards compatibility...